### PR TITLE
Feature: Supplementary files

### PR DESCRIPTION
**Pull request by oliver----**
## Why

See #1016, for instance.

## What

Supplementary files are represented as block-level elements, i.e. they can be placed in the manuscript body, or in an appendix, or in a figure caption. Intentionally we decided to allow this -- as it is defined in JATS -- as a general block-level element because this gives the author the most freedom in how to use these. E.g. they can be used anywhere in the manuscript body, or in a figure caption. Supplementary files can be referenced like other resources such as figures.

- [x] Extend the kitchen-sink (O)
- [x] Introduce SupplementaryFile model (O)
- [x] Add importer and exporter (O)
- [x] Implement label generator (simple) (D)
- [x] Implement UI Component (D)
- [x] Implement a command and tool for uploading and inserting supplementary file (D)
- [x] Ability to reference a supplementary file (D)
- [x] Ability to update an existing supplementary file (D)

Tests:
- [x] create a supplementary file (D)
- [x] remove a supplementary file (D)
- [x] reference a supplementary file (D)
  - [x] then insert another supplement before (reference should have updated)
  - [x] remove the referenced supplement (reference should be broken)
- [x] update an existing supplementary file
  - until we have better means to check if the file has actually changes, we should just check that the code does not throw

Not in this PR:
- Scoped labels, e.g. 'Figure 1 - Supplementary File 1.'

Mockups:

image

### CODE REVIEW

File: src/article/converter/r2t/SupplementaryFileConverter.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    if (labelEl) {
+      node.label = labelEl.text()
+    }
+    node.href = el.getAttribute('xlink:href')
+    node.legend = importer.convertElement(captionEl).id
+  }
+
+  _getContent (el) {
+    return findChild(el, 'tex-math')
```
Comment (Integral) [inline src/article/converter/r2t/SupplementaryFileConverter.js]: deadcode

Comment (oliver----) [reply src/article/converter/r2t/SupplementaryFileConverter.js]: Thx!

**Comment by Integral** [comment]
I think it's finished.

File: test/SupplementaryFile.test.js [hunk 2]
```diff
@@ -117,3 +117,26 @@ test('Supplementary File: reference a file', t => {
...
+  loadBodyFixture(editor, SUPPLEMENT_FILE)
+
+  const replaceSupplementaryFileToolSelector = '.sc-upload-supplementary-file-tool.sm-upload-tool'
+  editorSession.setSelection({
+    type: 'node',
+    nodeId: 'sm1',
+    surfaceId: 'body',
+    containerId: 'body'
+  })
+  const replaceSubFigureImageTool = editor.find(replaceSupplementaryFileToolSelector)
```
Comment (oliver----) [inline test/SupplementaryFile.test.js]: should be called 'replaceSupplementaryFileTool'

File: src/article/shared/UploadTool.js [hunk 3]
```diff
@@ -3,14 +3,17 @@ import { ToggleTool } from '../../kit'
...
-    const fileType = this.getFileType()
     const isMultiple = this.uploadMultiple()
     const input = $$('input').attr({
-      'type': 'file',
-      'accept': fileType
+      'type': 'file'
     }).ref('input')
       .on('change', this.onFileSelect)
 
+    if (!this.acceptAllFileTypes) {
```
Comment (oliver----) [inline src/article/shared/UploadTool.js]: This looks like a regression. Either define it as a property, or call it as a function

Comment (oliver----) [reply src/article/shared/UploadTool.js]: Spotted it, because accelAllFileTypes was never called.

**Pull request merged**